src/log.ts

```diff
@@ -24,3 +26,43 @@ export const configure = (debug: boolean = false): void => {
     }
   })
 }
+
+type CodecOperation = 'decode' | 'encode'
+
+export const errorNodeType = (
```

Comment:
I don't like the name of this function, I think it could be a little bit more verbose. Without referring back to this file it kind of loses context. Something like `logErrorNodeType` would be better.

Reply:
Sure. If we agree this "log all errors and warnings" approach is one we want to take then we should have good names for them, because we'll be using them a lot.

Reply:
Alternatively: write code that doesn't generate errors.